browser/brave_content_browser_client.cc

```diff
@@ -408,28 +412,37 @@ void BindBraveSearchDefaultHost(
 void BindIAPSubscription(
     content::RenderFrameHost* const frame_host,
     mojo::PendingReceiver<ai_chat::mojom::IAPSubscription> receiver) {
-  auto* context = frame_host->GetBrowserContext();
-  auto* profile = Profile::FromBrowserContext(context);
-  mojo::MakeSelfOwnedReceiver(
-      std::make_unique<ai_chat::AIChatIAPSubscription>(profile->GetPrefs()),
-      std::move(receiver));
+  const GURL frame_host_url = frame_host->GetLastCommittedURL();
+  if (skus::IsSafeOrigin(frame_host_url)) {
```

Comment:
nit: usually a flow like this is easier to keep track of:
```
if (!safe) {
  return;
}
```

there's no unnecessary nestedness.